### PR TITLE
Update TL User Guide

### PR DESCRIPTION
**Pull request by Scienfitz**
Updated with info and examples for when and when not to use it, feel free to suggest more examples

### CODE REVIEW

File: docs/userguide/transfer_learning.md [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
+previous campaigns can be learned, basically enabling us to utilize previous data. For
+instance, this can unlock data treasures coming from similar but not identical campaigns
+accumulated over many years.
+
+```{admonition} Expectations
+:class: important
+Because of the need to learn the covariance between tasks, transfer learning is not a
+magic method for zero-shot learning. It will always need data from an ongoing campaign
+to understand how other campaigns' data are related. Overall, if correlated task data
+were provided, the optimization of new campaigns can experience a dramatic speedup.
```
Comment (AVHopp) [inline docs/userguide/transfer_learning.md:56]: Would add a warning that learning can also be worse if you add stuff that is unrelated.

**Review by AVHopp — APPROVED** [review]

**Review by AdrianSosic — APPROVED** [review]
Hey @Scienfitz, thanks for writing the new parts. I did another iteration where I changed a few things and added even more info. Let me know if you have any concerns 👍🏼

File: docs/userguide/transfer_learning.md (hunk 1 above)
Comment (Scienfitz) [reply docs/userguide/transfer_learning.md]: added 
image
